src/EcalLumiSpecWScFi_geo.cpp

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2023 Aranya Giri
+
+/* Date : 12/12/2023
+W Scifi (EM Calorimeter) Pair Spectrometer */
+
+#include "DD4hep/DetFactoryHelper.h"
+#include <XML/Helper.h>
+#include <algorithm>
+#include <iostream>
+#include <tuple>
+#include <TVector3.h>
+
+using namespace std;
+using namespace dd4hep;
+
+// Definition of function to build the modules
+static tuple<Volume, Position> build_specScFiCAL_module(const Detector& description, const xml::Component& mod_x, SensitiveDetector& sens);
+
+// Driver Function
+static Ref_t create_detector(Detector& description, xml_h e, SensitiveDetector sens)
+{
+  sens.setType("calorimeter");
+
+  xml_det_t     x_det           =       e;
+  xml_comp_t    x_mod           =       x_det.child( _Unicode(module) );
+  string        det_name        =       x_det.nameStr();
+  int           det_ID          =       x_det.id();
+
+  // Create main detector element to be returned at the end
+  DetElement    det( det_name, det_ID );
+
+  // Mother volume
+  Volume        motherVol = description.pickMotherVolume( det );
+
+  // Detector assembly
+  Assembly      assembly( det_name );
+  assembly.setVisAttributes( description.invisible() );
+
+  double detSizeXY      = getAttrOrDefault( x_det, _Unicode(sizeXY), 180*mm);
+  double detSizeZ       = getAttrOrDefault( x_det, _Unicode(sizeZ), 180*mm);
+  int nmod_perlayer     = getAttrOrDefault( x_det, _Unicode(nmod_perlayer), 3);
+  int nlayer            = getAttrOrDefault( x_det, _Unicode(nlayer), 20);
```

Comment:
These are currently contained within the module rather than detector definition in the xml. Should swap to x_mod